### PR TITLE
Allow to run matchers on Document

### PR DESCRIPTION
**Pull request by sheerun**
**What**:

Allows to run matchers on document as so:

```es6
  const {window} = new JSDOM(
    `
    <!DOCTYPE html>
    <html lang="en" class="awesome" style="box-sizing: border-box">
      <head>
        <meta charset="utf-8">
      </head>
      <body id="lol">
        <div className="lol2">Hello world</div>
      </body>
    </html>
  `)

  expect(window.document).toContainElement(window.document.body)
  expect(window.document).toContainElement(window.document.head)
  expect(window.document).toContainHTML('<meta charset="utf-8">')
  expect(window.document).toHaveAttribute('lang', 'en')
  expect(window.document).toHaveClass('awesome')
  expect(window.document).toBeEnabled()
  expect(window.document).toHaveStyle('box-sizing: border-box')
  expect(window.document).toHaveTextContent('Hello world')
```

**Why**:

window.document is exceptional because its root DOM node in window.document.documentElement. This PR makes it possible to run matchers directly on window.document, shortening test code and allowing for easy and natural matching of `html` element

**How**:

I changed checkHtmlElement to getHtmlElement which returns htmlElement given any object that can matcher can run on. This is also a nice extension point for the future if you decide to accept any other objects for matchers.

**Checklist**:

* [x] Documentation
* [x] Tests
* [ ] Updated Type Definitions N/A
* [x] Ready to be merged 
* [ ] Added myself to contributors table N/A

### CODE REVIEW

**Comment by sheerun** [comment]
Forgot to tell: it's totally backward compatible

File: src/utils.js [hunk 1]
```diff
@@ -39,26 +39,37 @@ class HtmlElementTypeError extends Error {
...
+    return htmlElement.defaultView
   }
+
+  throw new HtmlElementTypeError(htmlElement, ...args)
 }
 
-function checkHtmlElement(htmlElement, ...args) {
-  checkHasWindow(htmlElement, ...args)
-  const window = htmlElement.ownerDocument.defaultView
+function getHtmlElement(htmlElement, ...args) {
```
Comment (eps1lon) [inline src/utils.js]: Why do we need to re-purpose this? This is now used to reassigning function arguments which is always a bit iffy. It's just the identity function but if it isn't necessary then why introduce it in the first place?

**Comment by sheerun** [comment]
You are not encouraging me to work on such PRs. I've submitted similar feature to to dom-testing-library and even it got merged, but some time later both tests and functionality has been dropped because somebody would consider it "bending over backwards"  https://github.com/testing-library/dom-testing-library/issues/100

If you want this functionality, please fix this PR yourselves, or close it like other one

**Comment by eps1lon** [comment]
> If you want this functionality, please fix this PR yourselves, or close it like other one

Ok I'll work on this myself then if you're not interested in collaborating. Thanks for reporting the issue.

**Pull request closed without merging**